src/vs/workbench/contrib/terminal/browser/media/shellIntegration-bash.sh

```diff
@@ -0,0 +1,63 @@
+IN_COMMAND_EXECUTION="1"
+prompt_start() {
+    printf "\033]133;A\007"
+}
+
+prompt_end() {
+    printf "\033]133;B\007"
+}
+
+update_cwd() {
+    printf "\033]1337;CurrentDir=%s\007" "$PWD"
+}
+
+command_output_start() {
+    printf "\033]133;C\007"
+}
+
+command_complete() {
+    printf "\033]133;D;%s\007" "$STATUS"
+    update_cwd
+}
+
+set_shell_integration_enabled() {
+    printf "\033]133;E\007"
+}
+
+update_prompt() {
+    PRIOR_PROMPT="$PS1"
+    IN_COMMAND_EXECUTION=""
+    local PREFIX=""
+    if [[ $PS1 == *"$(prompt_start)"* ]]; then
```

Comment:
It's not clear what this is doing? If `update_prompt` is called multiple times I'm not sure why?